### ISSUE TITLE
[BUG] Hides the blue bar for some videos on YT

### ISSUE THREAD

**Issue by ibehnam** [elliotwaite/thumbnail-rating-bar-for-youtube#73]
As seen in the picture, the blue bar doesn't appear for all videos. I've seen this issue in search results and the recommended videos on the right.

<img width="714" alt="image" src="https://user-images.githubusercontent.com/58621210/168452282-edf8867e-2dae-4957-b1dd-760a280fdf66.png">

**Comment by elliotwaite** [comment]
Thanks for reporting this. This is a known issue that usually results from hitting the rate limit of the Return YouTube Dislike API. I have a short explanation about it in the README [here](https://github.com/elliotwaite/thumbnail-rating-bar-for-youtube#rate-limiting-and-the-cache-duration-setting). You can usually try refreshing the page after about 5-15 seconds to retry retrieving the ratings for those videos, but I should really just update the extension to have it auto-retry failed requests on its own after a certain amount of time. I'll work on an update to add the auto-retry logic.

**Comment by elliotwaite** [comment]
I just released 1.8.3 which adds the auto-retry feature.

If an API request fails, it will sleep for 5 seconds and then retry the request. It will continue doing this for up to a max of 7 attempts for that thumbnail. After 7 attempts (~30 seconds) without success, a full page refresh would be required to retry retrieving the data for that thumbnail. Let me know if you think any of these settings should be changed.

Also, in my testing, it seemed like I was hitting the rate limit more than usual. So there might be something going on right now with the Return YouTube Dislike API server, not sure.

Let me know if this resolves this issue for you, or if you think there would be a better solution for this.

**Comment by elliotwaite** [comment]
I just updated the retry strategy in v1.8.5.

The problem with my initial strategy was that it would continue retrying to process failed thumbnails that had already been removed from the page.

In the new strategy, when an API request gets rate-limited (or fails for any other reason), it marks that thumbnail as being unprocessed and sets a timeout for 5-seconds later to retry processing any unprocessed thumbnails that are still visible on the page.

If a thumbnail cannot be successfully processed after 10 retries, it gives up on that thumbnail.
